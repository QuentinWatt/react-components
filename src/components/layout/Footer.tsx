const Footer: React.FC = () => {
  const date = new Date();

  return (
    <footer className="bg-gray-800 text-white">
      <div className="border-b border-gray-500">
        <div className="container mx-auto px-3">
          <div className="grid sm:grid-cols-2 lg:grid-cols-4 my-6">
            <ul>
              <li>A list item</li>
              <li>Second item</li>
              <li>Third item</li>
              <li>Fourth item</li>
            </ul>

            <ul>
              <li>A list item</li>
              <li>Another item</li>
              <li>Other item</li>
              <li>Last item</li>
            </ul>

            <ul>
              <li>A list item</li>
              <li>Second item</li>
              <li>Third item</li>
              <li>Fourth item</li>
            </ul>

            <div>
              <span></span>
            </div>
          </div>
        </div>
      </div>
      <div className="container mx-auto px-3 py-6">
        <div className="flex justify-between items-center">
          <span className="font-bold text-xl">Qcasts</span>

          <span className="text-xs">
            &copy; {date.getFullYear()} Quentin Watt
          </span>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
