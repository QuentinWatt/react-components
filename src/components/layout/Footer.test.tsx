import { render, screen } from "@testing-library/react";
import Footer from "./Footer";

describe("It has the correct footer", () => {
  it("has a copyright", () => {
    render(<Footer />);
    expect(screen.getByText("Â© 2024 Quentin Watt")).toBeTruthy();
    expect(screen.getByText("Qcasts")).toBeTruthy();
  });

  it("has the correct list items", () => {
    render(<Footer />);
    expect(screen.getAllByText("A list item")).toBeTruthy();
    expect(screen.getAllByText("Second item")).toBeTruthy();
    expect(screen.getAllByText("Third item")).toBeTruthy();
    expect(screen.getAllByText("Fourth item")).toBeTruthy();
  });
});
