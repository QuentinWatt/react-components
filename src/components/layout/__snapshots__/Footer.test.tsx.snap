// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It has the correct footer matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="bg-gray-800 text-white"
  >
    <div
      class="border-b border-gray-500"
    >
      <div
        class="container mx-auto px-3"
      >
        <div
          class="grid sm:grid-cols-2 lg:grid-cols-4 my-6"
        >
          <ul>
            <li>
              A list item
            </li>
            <li>
              Second item
            </li>
            <li>
              Third item
            </li>
            <li>
              Fourth item
            </li>
          </ul>
          <ul>
            <li>
              A list item
            </li>
            <li>
              Another item
            </li>
            <li>
              Other item
            </li>
            <li>
              Last item
            </li>
          </ul>
          <ul>
            <li>
              A list item
            </li>
            <li>
              Second item
            </li>
            <li>
              Third item
            </li>
            <li>
              Fourth item
            </li>
          </ul>
          <div>
            <span />
          </div>
        </div>
      </div>
    </div>
    <div
      class="container mx-auto px-3 py-6"
    >
      <div
        class="flex justify-between items-center"
      >
        <span
          class="font-bold text-xl"
        >
          Qcasts
        </span>
        <span
          class="text-xs"
        >
          Â© 2024 Quentin Watt
        </span>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
